### PR TITLE
refactor(plugins/response-transformer): code clean

### PR DESCRIPTION
**Pull request by chronolaw**
### Summary

Code clean and small optimization.

### Checklist

- [x] The Pull Request has tests
- [x] There's not an entry in the CHANGELOG
- [x] There is not a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Full changelog

* localize some functions
* use function `json_value` to simplify logic
* remove `is_json_body` in body_transformer
* move `is_json_body` test cases into header_transformer

### CODE REVIEW

File: kong/plugins/response-transformer/body_transformer.lua [hunk 1]
```diff
@@ -56,7 +78,7 @@ local function append_value(current_value, value)
   local current_value_type = type(current_value)
 
   if current_value_type  == "string" then
```
Comment (samugi) [inline kong/plugins/response-transformer/body_transformer.lua:80]: style nitpick: remove space before =
```suggestion
  if current_value_type == "string" then
```

File: kong/plugins/response-transformer/body_transformer.lua [hunk 2]
```diff
@@ -56,7 +78,7 @@ local function append_value(current_value, value)
...
-    return {current_value, value }
+    return { current_value, value }
   end
 
   if current_value_type  == "table" then
```
Comment (samugi) [inline kong/plugins/response-transformer/body_transformer.lua:84]: style nitpick: remove space before =
```suggestion
  if current_value_type == "table" then
```

File: kong/plugins/response-transformer/body_transformer.lua (hunk 1 above)
Comment (chronolaw) [reply kong/plugins/response-transformer/body_transformer.lua]: done, thanks.

File: kong/plugins/response-transformer/body_transformer.lua (hunk 2 above)
Comment (chronolaw) [reply kong/plugins/response-transformer/body_transformer.lua]: done, thanks.

**Review by samugi — APPROVED** [review]

**Review by liverpool8056 — APPROVED** [review]

**Pull request merged**